src/FreeDSx/Snmp/Protocol/ClientProtocolHandler.php

```diff
@@ -209,6 +209,7 @@ protected function performDiscovery(MessageRequestV3 $message, SecurityModelModu
         $id = $this->generateId();
         $this->setPduId($discovery->getRequest(), $id);
         $response = $this->sendRequestGetResponse($discovery);
+        assert($response instanceof MessageResponseV3);
```

Comment:
Could you throw a SecurityModelException here instead if null? Something like: Expected a discovery response, but none was received.

Reply:
What about now?